src/udiskslinuxfilesystem.c

```diff
@@ -1680,6 +1680,42 @@ get_error_code_for_umount (const gchar *error_message)
     return UDISKS_ERROR_FAILED;
 }
 
+typedef struct
+{
+  const gchar *object_path;
+  guint64      old_size;
+  gchar       *mount_point;
+} WaitForFilesystemMountPointsData;
+
+static UDisksObject *
+wait_for_filesystem_mount_points (UDisksDaemon *daemon,
+                                  gpointer      user_data)
+{
+  WaitForFilesystemMountPointsData *data = user_data;
+  UDisksObject *object = NULL;
+  UDisksFilesystem *filesystem = NULL;
+  const gchar * const *mount_points = NULL;
+
+  object = udisks_daemon_find_object (daemon, data->object_path);
+
+  if (object != NULL)
+    {
+      filesystem = udisks_object_peek_filesystem (object);
+      if (filesystem != NULL)
+        {
+          mount_points = udisks_filesystem_get_mount_points (filesystem);
+        }
+      /* If we know which mount point should have gone, directly test for it, otherwise test if any has gone */
+      if (mount_points == NULL || (data->mount_point != NULL && g_strv_contains (mount_points, data->mount_point)) ||
+          g_strv_length ((gchar **) mount_points) == data->old_size)
```

Comment:
This test is racy - somebody may have mounted the filesystem somewhere else (or multiple times) in the meantime. Probably not worth testing this.

Reply:
Through UDisks the filesystem cannot be mounted/unmounted because the Mutex is hold. Should we consider users/daemons that do directly mount/unmount without UDisks? Since the worst case here is a timeout, I think this condition should be kept. But as it already covers a special case, I can also remove the length check.

Reply:
Yes, there's a chance something external could mess with the block device and mount it somewhere else (given the filesystem supports multiple mounts). The worst case is reaching the timeout, right, and it wouldn't fail the whole operation, so it's not that critical. Alright, let's keep it there for the time being unless it causes any issue.